cirq/testing/equals_tester.py

```diff
@@ -78,8 +83,13 @@ def add_equality_group(self, *group_items: Any):
                 assert hasattr(v1, '__eq__') == hasattr(v1, '__ne__')
                 # Careful: python2 int doesn't have __eq__ or __ne__.
                 if hasattr(v1, '__eq__'):
-                    eq = v1.__eq__(v2)
-                    ne = v1.__ne__(v2)
+                    eq = NotImplemented
+                    ne = NotImplemented
+                    try:
+                        eq = v1.__eq__(v2)
+                        ne = v1.__ne__(v2)
+                    except TypeError:
```

Comment:
This should never ever be happening! Don't ignore this error, it is an actual failure!

Reply:
Actually it only happens if one tries to add sequence of something like (object, object).
<pre>&gt;&gt;&gt; a = object
&gt;&gt;&gt; b = object
&gt;&gt;&gt; a &lt; b
Traceback (most recent call last):
  File &quot;&lt;stdin&gt;&quot;, line 1, in &lt;module&gt;
TypeError: unorderable types: type() &lt; type()
</pre>

Reply:
Digged up the actual error from the logs before that change:

```
______________________ test_add_ordering_group_incorrect _______________________

    def test_add_ordering_group_incorrect():
        ot = OrderTester()
        ot.add_ascending(1, 2)
        with pytest.raises(AssertionError):
>           ot.add_ascending(object, object)  # not ascending within call

cirq/testing/order_tester_test.py:31: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
cirq/testing/order_tester.py:148: in add_ascending
    self.add_ascending_equivalence_group(item)
cirq/testing/order_tester.py:166: in add_ascending_equivalence_group
    super(OrderTester, self).verify_equality_group(*group_items)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <cirq.testing.order_tester.OrderTester object at 0x7fa71a6ca128>
group_items = (<class 'object'>,), v1 = <class 'object'>, v2 = <class 'object'>

    def verify_equality_group(self, *group_items: Any):
        """Verifies that a group is an equivalence group.
    
            This methods asserts that items within the group must all be equal to
            each other, but not equal to any items in other groups that have been
            or will be added.
    
            Args:
              *group_items: The items making up the equivalence group.
    
            Raises:
                AssertionError: Items within the group are not equal to each other,
                    or items in another group are equal to items within the new
                    group, or the items violate the equals-implies-same-hash rule.
            """
    
        assert group_items
    
        # Check that group items are equivalent to each other.
        for v1, v2 in itertools.product(group_items, group_items):
            # Binary operators should always work.
            assert v1 == v2
            assert not v1 != v2
    
            # __eq__ and __neq__ should both be correct or not implemented.
            assert hasattr(v1, '__eq__') == hasattr(v1, '__ne__')
            # Careful: python2 int doesn't have __eq__ or __ne__.
            if hasattr(v1, '__eq__'):
>               eq = v1.__eq__(v2)
E               TypeError: expected 1 arguments, got 0

cirq/testing/equals_tester.py:64: TypeError
```

Reply:
That seems to be a known issue for class objects: https://bugs.python.org/issue10017

Reply:
It looks like this is a perfect reason not to use the `__eq__` and friends directly as I've been doing so far in the tests. I've fixed `EqualsTester` to only use `==` and `!=` instead of worrying about `__eq__`, and we should the same here.